tokio-native-tls/src/lib.rs

```diff
@@ -362,13 +362,16 @@ impl<S: AsyncRead + AsyncWrite + Unpin> Future for MidHandshake<S> {
 
         s.get_mut().context = cx as *mut _ as *mut ();
         match s.handshake() {
-            Ok(stream) => Poll::Ready(Ok(TlsStream(stream))),
-            Err(HandshakeError::Failure(e)) => Poll::Ready(Err(e)),
+            Ok(mut s) => {
+                s.get_mut().context = null_mut();
```

Comment:
So referencing back to your comment, it looks like in this case we return the stream where I assume we want the context to be set and not null?

Reply:
Do you mean comment from original PR? When I checked how tls implemented I found difference in result between `StartedHandshakeFuture` and `MidHandshake`. Why we should return stream with context not equal `null_mut`? If I understand correctly, after `poll` call pointer to context will be invalid? That's why `Guard` struct exists.

Reply:
Yeah, I see that now.